### PR TITLE
Release 3.0.0

### PR DESCRIPTION
**Pull request by smk4664**
## What's Changed

Add support for Nautobot 2.0
Add ability to link Chat Platform User accounts to Nautobot User

## TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/chatops/en/latest/dev/contributing/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by jathanism — COMMENTED** [review]
Very clean. No gripes! :shipit:

File: nautobot_chatops/admin.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """Administrative capabilities for nautobot_chatops plugin."""
 
 from django.contrib import admin
-from .models import CommandLog
+from .models import CommandLog, ChatOpsAccountLink
```
Comment (HanlinMiao) [inline nautobot_chatops/admin.py:4]:
```suggestion
from .models import ChatOpsAccountLink, CommandLog
```

File: nautobot_chatops/api/views/generic.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 """API Views for Nautobot Chatops."""
 from rest_framework.routers import APIRootView
-from nautobot.core.api.views import ModelViewSet
+from nautobot.apps.api import NautobotModelViewSet
 
-from nautobot_chatops.api.serializers import AccessGrantSerializer, CommandTokenSerializer
-from nautobot_chatops.models import AccessGrant, CommandToken
-from nautobot_chatops.filters import AccessGrantFilterSet, CommandTokenFilterSet
+from nautobot_chatops.api.serializers import AccessGrantSerializer, CommandLogSerializer, CommandTokenSerializer
+from nautobot_chatops.models import AccessGrant, CommandToken, CommandLog
```
Comment (HanlinMiao) [inline nautobot_chatops/api/views/generic.py:6]:
```suggestion
from nautobot_chatops.models import AccessGrant, CommandLog, CommandToken
```

File: nautobot_chatops/api/views/generic.py [hunk 3]
```diff
@@ -15,15 +15,23 @@ def get_view_name(self):
         return "Nautobot Chatops"
 
 
-class CommandTokenViewSet(ModelViewSet):  # pylint: disable=too-many-ancestors
+class CommandTokenViewSet(NautobotModelViewSet):  # pylint: disable=too-many-ancestors
```
Comment (HanlinMiao) [inline nautobot_chatops/api/views/generic.py:18]:
```suggestion
class CommandTokenViewSet(NautobotModelViewSet):
```

File: nautobot_chatops/api/views/generic.py [hunk 4]
```diff
@@ -15,15 +15,23 @@ def get_view_name(self):
...
-class AccessGrantViewSet(ModelViewSet):  # pylint: disable=too-many-ancestors
+class CommandLogViewSet(NautobotModelViewSet):
+    """API viewset for interacting with CommandLog objects."""
+
+    queryset = CommandLog.objects.all()
+    serializer_class = CommandLogSerializer
+    filterset_class = CommandLogFilterSet
+
+
+class AccessGrantViewSet(NautobotModelViewSet):  # pylint: disable=too-many-ancestors
```
Comment (HanlinMiao) [inline nautobot_chatops/api/views/generic.py:34]:
```suggestion
class AccessGrantViewSet(NautobotModelViewSet):
```

File: nautobot_chatops/api/views/ms_teams.py [hunk 5]
```diff
@@ -139,6 +139,7 @@ def post(self, request, *args, **kwargs):
             "user_id": body["from"]["id"],
             "user_name": body["from"]["name"],
             "user_role": body["from"].get("role"),
+            "user_ad_id": body["from"].get("aadObjectId"),
```
Comment (HanlinMiao) [inline nautobot_chatops/api/views/ms_teams.py:142]: Just want to make sure `aadObjectId` is correct here

File: nautobot_chatops/filters.py [hunk 6]
```diff
@@ -1,8 +1,10 @@
 """Django FilterSet classes for Nautobot."""
 
-from nautobot.utilities.filters import BaseFilterSet
+import django_filters
+from nautobot.apps.filters import BaseFilterSet, NautobotFilterSet, SearchFilter
 
-from nautobot_chatops.models import CommandLog, AccessGrant, CommandToken
+from nautobot_chatops.choices import PlatformChoices
+from nautobot_chatops.models import CommandLog, AccessGrant, ChatOpsAccountLink, CommandToken
```
Comment (HanlinMiao) [inline nautobot_chatops/filters.py:7]:
```suggestion
from nautobot_chatops.models import AccessGrant, ChatOpsAccountLink, CommandLog, CommandToken
```

File: nautobot_chatops/forms.py [hunk 7]
```diff
@@ -2,10 +2,11 @@
 
 from django import forms
 
-from nautobot.utilities.forms import BootstrapMixin
+from nautobot.core.forms import BootstrapMixin, StaticSelect2Multiple
+from nautobot.extras.forms import NautobotFilterForm
 
-from .models import AccessGrant, CommandToken
-from .choices import AccessGrantTypeChoices, CommandTokenPlatformChoices
+from .models import AccessGrant, CommandLog, CommandToken, ChatOpsAccountLink
```
Comment (HanlinMiao) [inline nautobot_chatops/forms.py:8]:
```suggestion
from .models import AccessGrant, ChatOpsAccountLink, CommandLog, CommandToken
```

File: nautobot_chatops/templates/nautobot_chatops/chatops_account_link_edit.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+$(document).ready(function() {
+    $("#user-id-lookup").click(function() {
+        let platform = $("#id_platform").val();
+        if (!platform) {
+            alert("Please select a platform");
+            return;
+        }
+        let email = $("#id_email").val();
+        if (!email) {
+            alert("Please enter an email to look up the value for");
```
Comment (HanlinMiao) [inline nautobot_chatops/templates/nautobot_chatops/chatops_account_link_edit.html:33]: Im not sure how to make this sound better, perhaps:
```suggestion
            alert("Please enter an email to look up your User ID");
```

File: nautobot_chatops/tests/test_views.py [hunk 9]
```diff
@@ -16,7 +16,7 @@
     generate_signature as webex_generate_signature,
 )
 from nautobot_chatops.api.views.mattermost import verify_signature as mattermost_verify_signature
-from nautobot_chatops.choices import CommandTokenPlatformChoices
+from nautobot_chatops.choices import PlatformChoices
 from nautobot_chatops.models import CommandToken
 from nautobot_chatops.choices import CommandStatusChoices
```
Comment (HanlinMiao) [inline nautobot_chatops/tests/test_views.py:21]:
```suggestion
from nautobot_chatops.choices import CommandStatusChoices, PlatformChoices
from nautobot_chatops.models import CommandToken
```

File: nautobot_chatops/urls.py [hunk 10]
```diff
@@ -33,14 +37,32 @@
     logger.warning("Grafana ChatOps integration is not available.", exc_info=True)
 
 urlpatterns = [
-    path("", NautobotHomeView.as_view(), name="home"),
+    path("", CommandLogListView.as_view(), name="commandlog_list"),
```
Comment (HanlinMiao) [inline nautobot_chatops/urls.py:40]: Should this be?
```suggestion
    path("commandlog/", CommandLogListView.as_view(), name="commandlog_list"),
```

File: nautobot_chatops/views.py [hunk 11]
```diff
@@ -5,124 +5,50 @@
...
+        }
 
 
 class AccessGrantListView(PermissionRequiredMixin, ObjectListView):
     """View for listing all extant AccessGrants."""
 
     # Set the action buttons to correspond to what there are views. If import/export are added, this should be updated
     action_buttons = ("add",)
-    permission_required = "nautobot_chatops.view_accessgrant"
+    permission_required = "nautobot_chatops.view_commandlog"
```
Comment (HanlinMiao) [inline nautobot_chatops/views.py:47]: Is this change correct here? Seems like the old one is more correct.

File: nautobot_chatops/views.py [hunk 12]
```diff
@@ -5,124 +5,50 @@
...
-    return f"{prefix}.{deciseconds}"
-
-
-class NautobotHomeView(PermissionRequiredMixin, View):
-    """Homepage view for Nautobot."""
-
-    permission_required = "nautobot_chatops.view_commandlog"
-    http_method_names = ["get", "post"]
+class CommandLogListView(PermissionRequiredMixin, ObjectListView):
+    """View for listing all extant AccessGrants."""
```
Comment (HanlinMiao) [inline nautobot_chatops/views.py:26]:
```suggestion
    """View for listing all extant Command Logs."""
```

File: nautobot_chatops/views.py [hunk 13]
```diff
@@ -5,124 +5,50 @@
...
 
+    action_buttons = ("export",)
+    permission_required = "nautobot_chatops.view_accessgrant"
```
Comment (HanlinMiao) [inline nautobot_chatops/views.py:29]:
```suggestion
    permission_required = "nautobot_chatops.view_commandlog"
```

File: nautobot_chatops/views.py (hunk 11 above)
Comment (HanlinMiao) [inline nautobot_chatops/views.py:47]:
```suggestion
    permission_required = "nautobot_chatops.view_accessgrant"
```

File: nautobot_chatops/api/views/ms_teams.py (hunk 5 above)
Comment (smk4664) [reply nautobot_chatops/api/views/ms_teams.py:142]: Yes it is

File: nautobot_chatops/urls.py (hunk 10 above)
Comment (smk4664) [reply nautobot_chatops/urls.py:40]: Nope, this replaces that HomeView, kept in to keep the same feel from previous version.

**Pull request merged**